### ISSUE TITLE
login_required on silent callback

### ISSUE THREAD

**Issue by rickyoleary** [AxaFrance/oidc-client#1097]
Hi. I am using your React Vite demo project to test our server.

We have an ASP.NET Core server handling the OIDC.

On http://localhost:3000 it works fine. But on an IP, in this case I'm testing http://127.0.0.1 it logs in but the silent renew fails. When clicking "renew tokens" I get the following after console logging:
http://127.0.0.1:3000/authentication/silent_callback?error=login_required&state=CmZZeGfFWpS1djhU#_=_

I can see from the code when debugging and the OIDC console that the "code" is being passed as "undefined".

code=undefined&grant_type=authorization_code&client_id=OrchardMobile&redirect_uri=http%3A%2F%2F127.0.0.1%3A3000%2Fauthentication%2Fsilent_callback&code_verifier=ouydUAdMNc600A2qbS20uzY79xGLJWRgyrQ2v6rbzpZLcu28fWAmOBNahP3BvlI1CuMFko9eteMdEys9Vr22DrEXm1pw5nf4wXUO5ksS4MEAG3099sN2442QQCYaKwAI

Our CORS in the database are reflected to use the IP address, so I'm at a bit of a loss why this would happen? Any help appreciated. Thanks.

**Comment by guillaume-chervet** [comment]
hi @rickyoleary ,

Thank you for your issue.
Are you using Safari?

Silent sigiining wil work only 
if OIDC server configure cookie in a specific way.
or
if your application is in the same url domain than OIDC server URL.

**Comment by rickyoleary** [comment]
Hi, yes, the issue was although the OIDC server was running locally, I had to specify 127.0.0.1 for my client to work. If I explicitly hosted OIDC on localhost, localhost would work. In short, it was a domain/cross domain issue.

I did also encounter [this issue](https://github.com/AxaFrance/oidc-client/issues/1028) on your newest version, and like the user suggested, I reverted down to version 6.12.1 to fix this issue.
